tests/e2e/redis/basic-ingest-redis-serving.py

```diff
@@ -741,6 +741,77 @@ def test_list_entities_and_features(client):
     assert set(filter_by_project_entity_expected) == set(filter_by_project_entity_actual)
     assert set(filter_by_project_labels_expected) == set(filter_by_project_labels_actual)
 
+
+@pytest.fixture(scope='module')
+def list_type_dataframe():
+    N_ROWS = 2
+    time_offset = datetime.utcnow().replace(tzinfo=pytz.utc)
+    customer_df = pd.DataFrame(
+        {
+            "datetime": [time_offset] * N_ROWS,
+            "customer_id": [i for i in range(N_ROWS)],
+            "rating": [i for i in range(N_ROWS)],
+            "cost": [float(i)+0.5 for i in range(N_ROWS)],
+            "past_transactions_int": [[i,i+2] for i in range(N_ROWS)],
+            "past_transactions_double": [[float(i)+0.5,float(i)+2] for i in range(N_ROWS)],
+            "past_transactions_float": [[float(i)+0.5,float(i)+2] for i in range(N_ROWS)],
+            "past_transactions_string": [['first_'+str(i),'second_'+str(i)] for i in range(N_ROWS)],
+            "past_transactions_bool": [[True,False] for _ in range(N_ROWS)]
+        }
+    )
+    return customer_df
+
+
+@pytest.mark.timeout(600)
+@pytest.mark.run(order=43)
+def test_basic_retrieve_online_dict(client, list_type_dataframe):
+    customer_fs = FeatureSet(
+        name="customer2",
+        features=[
+            Feature(name="rating", dtype=ValueType.INT64, labels={"key1":"val1"}),
+            Feature(name="cost", dtype=ValueType.FLOAT),
+            Feature(name="past_transactions_int", dtype=ValueType.INT64_LIST),
+            Feature(name="past_transactions_double", dtype=ValueType.DOUBLE_LIST),
+            Feature(name="past_transactions_float", dtype=ValueType.FLOAT_LIST),
+            Feature(name="past_transactions_string", dtype=ValueType.STRING_LIST),
+            Feature(name="past_transactions_bool", dtype=ValueType.BOOL_LIST)
+        ],
+        entities=[Entity("customer_id", ValueType.INT64)],
+        max_age=Duration(seconds=600)
+    )
+
+    client.set_project(PROJECT_NAME)
+    client.apply(customer_fs)
+
+    customer_fs = client.get_feature_set(name="customer2")
+    client.ingest(customer_fs, list_type_dataframe, timeout=300)
+    time.sleep(15)
+
+    online_request_entity = [{"customer_id": 1}]
```

Comment:
Can we try multiple entities please and we also need to test failure, what if mixed types are provided or mixed list types.